R/analyse_pseudo_bulk.R

```diff
@@ -0,0 +1,331 @@
+  assay_data_aggregated <- as.data.frame(SingleCellExperiment::assay(aggregated_object))
+  meta_data_aggregated <- SingleCellExperiment::colData(aggregated_object)[,c(group_by,k_variable)]
+  
+  clustering_level <- meta_data_aggregated[[group_by]]
+  variable_pools <- meta_data_aggregated[[k_variable]]
+  
+  combined_list <- mapply(function(x, y) paste(x, y, sep = "_"), variable_pools, clustering_level)
+  combined_list <- as.list(combined_list)
+  colnames(assay_data_aggregated) <- combined_list
+  
+  return(assay_data_aggregated)
+}
+
+
+
+#' split SCE Object with random pooling
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param k_variable    number of pools that should be created
+#' 
+#' @returns             returns pseudo bulk generated data
+#' 
+#' @importFrom SingleCellExperiment scuttle
+#' 
+#' @examples
+#' # SCE_RESULT_RANDOM <- split_random_sce(SCE_OBJECT, GROUP_BY, K_NUMBER)
+#' 
+split_random_sce <- function(sce_object, group_by, k_variable){
+  metadata <- SingleCellExperiment::colData(sce_object)
+  
+  num_cells <- ncol(sce_object)
+  random_data <- sample(1:k_variable, num_cells, replace = TRUE)
+  SingleCellExperiment::colData(sce_object)$random_column <- random_data
+  
+  aggregated_counts <- scuttle::aggregateAcrossCells(sce_object, ids=colData(sce_object)[,c(group_by, "random_column")])
+  
+  meta_data_aggregated <- SingleCellExperiment::colData(aggregated_counts)[,c(group_by,"random_column")]
+  aggregated_counts <- as.data.frame(SingleCellExperiment::assay(aggregated_counts))
+  
+  clustering_level <- meta_data_aggregated[[group_by]]
+  random_pools <- meta_data_aggregated$random_column
+  
+  combined_list <- mapply(function(x, y) paste(x, y, sep = "_"),clustering_level, random_pools)
+  combined_list <- as.list(combined_list)
+  colnames(aggregated_counts) <- combined_list
+  
+  return(aggregated_counts)
+}
+
+
+
+#' split SCE Object with random pooling
+#' @param group_by      entry in metadata table, based on these cluster annotation pseudo bulk is performed
+#' @param k_variable    subcluster paramters
+#' 
+#' @param resolution    resolution
+#' @param cluster1      cluster to subcluster as areference
+#' @param cluster2      cluster to subcluster for comparison
+#' 
+#' @returns             returns pseudo bulk generated data
+#' 
+#' @importFrom SingleCellExperiment scuttle scran
+#' 
+#' @examples
+#' K_CLUSTER <- list(3, 1,4)
+#' # SCE_RESULT_CLUSTERING <- split_subclustering(SCE_OBJECT, 
+#'                                                nn.clusters, 
+#'                                                K_CLUSTER)
+#' 
+split_subclustering_sce <- function(sce_object, group_by, k_variable){
+  print(k_variable)
+  
+  #check if Dim reduction and Clustering is performed 
+  if(length(SingleCellExperiment::reducedDimNames(sce_object)) == 0){
+    print("No Dimensionaities for Subclustering available")
```

Comment:
Generally, refrain from using `print`. Here, I guess it's an error, so you should exist with en error message using `stop`